cmd/otel-allocator/config/config.go

```diff
@@ -109,9 +109,10 @@ func LoadFromCLI(target *Config, flagSet *pflag.FlagSet) error {
 		return err
 	}
 
-	target.PrometheusCR.Enabled, err = getPrometheusCREnabled(flagSet)
-	if err != nil {
-		return err
+	if prometheusCREnabled, changed, flagErr := getPrometheusCREnabled(flagSet); flagErr != nil {
```

Comment:
Note: This needs to be fixed for other flags as well, here I limited the fix to the relevant one.